mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -235,6 +245,8 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
 
     // Track if checkpoint exists for UI updates
     const [hasCheckpoint, setHasCheckpoint] = useState<boolean>(false);
+    const [notebookSnapshot, setNotebookSnapshot] = useState<AIOptimizedCell[] | null>(null);
```

Comment:
A few notes here: 

1. Can we call this `notebookSnapshotPreAgentExecution` and then maybe `notebookSnapshotAfterAgentExecution`? I wasn't sure what notebookSnapshot referred to
2. Do we need notebookSnapshotPreAgentExecution to be a state variable? instead, can it just be a ref?